wasm/execution/instructions.py

```diff
@@ -18,10 +18,15 @@ class InstructionSequence(Sequence):
     Stateful stream of instructions for web assembly execution.
     """
     _instructions: Tuple[BaseInstruction, ...]
+    current: BaseInstruction
 
     def __init__(self, instructions: Iterable[BaseInstruction]) -> None:
         self._instructions = tuple(instructions)
         self._idx = -1
+        try:
+            self.current = self._instructions[0]
+        except IndexError:
```

Comment:
If instructions is empty in some normal cases, it seems like it must be faster to do:

```py
if self._instructions:
  self.current = self._instructions[0]
```

If it is never supposed to happen, then why squash the exception?

Reply:
I've actually removed the memoization of current instruction from the sequence object and moved it down into the `Configuration` object which removes the need to check for this edge case.

Reply:
![stock-vector-cartoon-brain-having-a-brain-fart-vector-clip-art-illustration-with-simple-gradients-all-in-a-151038323](https://user-images.githubusercontent.com/824194/53762223-c1d3d000-3e84-11e9-8979-d58b3b9f11e4.jpg)